app/src/test/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmServicesTest.kt

```diff
@@ -0,0 +1,97 @@
+package nerd.tuxmobil.fahrplan.congress.alarms
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import nerd.tuxmobil.fahrplan.congress.contract.BundleKeys
+import nerd.tuxmobil.fahrplan.congress.models.SchedulableAlarm
+import org.junit.Test
+
+class AlarmServicesTest {
+
+    private var alarmManager = mock<AlarmManager>()
+    private var mockContext = mock<Context>()
+    private var pendingIntent = mock<PendingIntent>()
+    private val alarm = SchedulableAlarm(3, "1001", "Welcome", 700)
+
+    @Test
+    fun `scheduleEventAlarm invokes "cancel" then "set" when "discardExisting" is "true"`() {
+        val onPendingIntentBroadcast: PendingIntentCallback = { context, requestCode, intent, flags ->
+            assertThat(context).isEqualTo(mockContext)
+            assertThat(requestCode).isEqualTo(alarm.eventId.toInt())
+            assertIntentExtras(intent, AlarmReceiver.ALARM_LECTURE)
+            assertThat(flags).isEqualTo(0)
+            pendingIntent
+        }
+        val alarmServices = AlarmServices(alarmManager, onPendingIntentBroadcast)
+        alarmServices.scheduleEventAlarm(mockContext, alarm, true)
+        verify(alarmManager, once()).cancel(pendingIntent)
+        verify(alarmManager, once()).set(AlarmManager.RTC_WAKEUP, alarm.startTime, pendingIntent)
+    }
+
+    @Test
+    fun `scheduleEventAlarm only invokes "set" when "discardExisting" is "false"`() {
+        val onPendingIntentBroadcast: PendingIntentCallback = { context, requestCode, intent, flags ->
+            assertThat(context).isEqualTo(mockContext)
```

Comment:
How do you assert that onPendingIntentBroadcast has ever been called (and therefore these assertions are executed)?

Reply:
Good point. I have to think about it. If you have an idea how to verify the invocation please shoot.

Reply:
The `onPendingIntentBroadcast` invocation is verified indirectly by `verify(alarmManager, once()).set(AlarmManager.RTC_WAKEUP, alarm.startTime, pendingIntent)` which checks if the `pendingIntent` object matches the one returned by `onPendingIntentBroadcast`. Do you agree?